neqo-transport/src/connection/mod.rs

```diff
@@ -1795,9 +1795,20 @@ impl Connection {
 
         recv_streams_write_frames(&mut self.recv_streams, builder, tokens, stats)?;
 
-        self.flow_mgr
-            .borrow_mut()
-            .write_frames(builder, tokens, stats)?;
+        self.remote_streams_fc[StreamType::BiDi].write_frames(builder, tokens, stats)?;
+        if builder.remaining() < 2 {
+            return Ok(());
+        }
+        self.remote_streams_fc[StreamType::UniDi].write_frames(builder, tokens, stats)?;
+        if builder.remaining() < 2 {
+            return Ok(());
+        }
+
+        self.local_streams_fc[StreamType::BiDi].write_frames(builder, tokens, stats)?;
+        if builder.remaining() < 2 {
+            return Ok(());
+        }
+        self.local_streams_fc[StreamType::UniDi].write_frames(builder, tokens, stats)?;
         if builder.remaining() < 2 {
             return Ok(());
         }
```

Comment:
If you do as I suggest, then this block of code can be moved to the new StreamLimits object.